### PR TITLE
[MINOR][SQL] Fix comments style and correct several styles and nits in CSV data source

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

While trying to create a PR (which was not an issue at the end), I just corrected some style nits.

So, I removed the changes except for some coding style corrections.
- According to the [scala-style-guide#documentation-style](https://github.com/databricks/scala-style-guide#documentation-style), Scala style comments are discouraged.

> ``` scala
> /** This is a correct one-liner, short description. */
> 
> /**
>  * This is correct multi-line JavaDoc comment. And
>  * this is my second line, and if I keep typing, this would be
>  * my third line.
>  */
> 
> /** In Spark, we don't use the ScalaDoc style so this
>   * is not correct.
>   */
> ```
- Double newlines between consecutive methods was removed. According to [scala-style-guide#blank-lines-vertical-whitespace](https://github.com/databricks/scala-style-guide#blank-lines-vertical-whitespace), single newline appears when

> Between consecutive members (or initializers) of a class: fields, constructors, methods, nested classes, static initializers, instance initializers.
- Remove uesless parentheses in tests 
- Use `mapPartitions` instead of `mapPartitionsWithIndex()`.
## How was this patch tested?

Unit tests were used and `dev/run_tests` for style tests.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVRelation.scala [hunk 1]
```diff
@@ -42,12 +42,12 @@ object CSVRelation extends Logging {
       firstLine: String,
       params: CSVOptions): RDD[Array[String]] = {
     // If header is set, make sure firstLine is materialized before sending to executors.
-    file.mapPartitionsWithIndex({
-      case (split, iter) => new BulkCsvReader(
+    file.mapPartitions { iter =>
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVRelation.scala]: This is legitimately a little optimization, but the rest feels too minor to bother with. Really there are more instances of slightly mis-spaced scaladoc or lines of code, though to fix them all would be time-consuming. Fixing them when nearby code changes is sensible. Fixing a few at random doesn't feel like a great goal.

Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVRelation.scala]: I see. Thanks! Then, should I close this or just get rid of comment changes here?

Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVRelation.scala]: Oh, actually I noticed that there are not so many of them. I can change all the comments and maybe I could add a pattern.

**Comment by rxin** [comment]
The doc changes are actually good. Going to merge this in master. Thanks.

**Pull request closed without merging**